### DISCUSSION TITLE
How to open a collaboration link within a PWA?

### DISCUSSION THREAD

**Discussion by sojusnik** [excalidraw/excalidraw#2712] [Q&A]
I mainly use Excalidraw as a PWA created with [Brave](https://en.wikipedia.org/wiki/Brave_(web_browser)). Since a PWA has no URL bar, I can't paste a link from a previous collaboration session in order to continue working on it. Starting a new collaboration session though is possible by clicking on the collaboration icon.

Any ideas how to open an existing collaboration link directly in a PWA?

**Comment by dwelle** [comment]
Interesting. I haven't thought of this use case. @tomayac what's the preferred solution for this? Should apps be implementing their own url bars? :p — in our case, should we add a url input inside collab dialog? Maybe just for PWAs? If so, is `window.matchMedia('(display-mode: standalone)').matches` the authoritative way to check if in a PWA?

@sojusnik this is the [exact case](https://github.com/excalidraw/excalidraw/discussions/2715#discussioncomment-261194) where I'd like to make this discussion into an issue, and cant :smile:

**Reply by tomayac** [reply]
> Interesting. I haven't thought of this use case. @tomayac what's the preferred solution for this? Should apps be implementing their own url bars? :p — in our case, should we add a url input inside collab dialog?

In this case I would probably expect a popup dialog that asks the user to enter the "secret" (like the session ID), and that silently also just accepts a full URL. 

> Maybe just for PWAs? If so, is `window.matchMedia('(display-mode: standalone)').matches` the authoritative way to check if in a PWA?

`window.matchMedia('(display-mode: standalone)').matches` is the correct way, yes.

**Reply by lipis** [reply]
> @sojusnik this is the [exact case](https://github.com/excalidraw/excalidraw/discussions/2715#discussioncomment-261194) where I'd like to make this discussion into an issue, and cant 😄

@dwelle For now let's file it manually until GitHub will make up their mind.. I think it will be tricky, because here you can reply inside a comment :)

**Reply by sojusnik** [reply]
Do we already have a solution how to open an URL (for collaboration) from a PWA?
